### PR TITLE
Improved Logger

### PR DESCRIPTION
**Pull request by kglebows**
Hello! 
Ive added a define macro for log messages that u can pass whatever ostringstream can take in! 
Also added Seconds for timestamp, and added File and Line in log messages.
image

image

Also added hide function that after Logger initialization it filters out specific level
image

image

### CODE REVIEW

File: src/config-manager/ConfigManager.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@ void ConfigManager::parseConfigFile(std::string configFilePath) {
     }
     configFile.close();
     validateConfiguration();
-    Logger::log("Parsing was successful", Logger::SUCCESS);
+    // Logger::log("Parsing was successful", Logger::SUCCESS);
```
Comment (RealConrad) [inline src/config-manager/ConfigManager.cpp]: Remove this line

File: src/logger/Logger.cpp [hunk 2]
```diff
@@ -57,7 +85,7 @@ void Logger::setLogFile(const std::string& filename) {
 
     fileStream.open(filename.c_str());
     if (fileStream.fail()) {
-        log("Failed to open log file: " + filename, ERROR);
+        // ERROR("Failed to open log file: " + filename);
```
Comment (RealConrad) [inline src/logger/Logger.cpp]: Change this to use the new Macro style of logger

File: src/logger/Logger.cpp [hunk 3]
```diff
@@ -70,20 +98,24 @@ std::string Logger::getLevelString(LogLevel level) {
...
-            log("UNKNOWN LOG LEVEL!", ERROR);
+            // log("UNKNOWN LOG LEVEL!", ERROR);
             return "UNKNOWN";
     }
 }
 
 std::string Logger::getColor(LogLevel level) {
     switch (level) {
+        case DEBUG:
+            return "\033[38;5;208m"; // Attempt for a less intense green (might look the same as regular green)
```
Comment (RealConrad) [inline src/logger/Logger.cpp]: Remove long comment, just give the colour name

File: src/logger/Logger.cpp [hunk 4]
```diff
@@ -70,20 +98,24 @@ std::string Logger::getLevelString(LogLevel level) {
...
+            return "DEBUG";
         case INFO:
             return "INFO";
         case WARNING:
             return "WARNING";
         case ERROR:
             return "ERROR";
         default:
-            log("UNKNOWN LOG LEVEL!", ERROR);
+            // log("UNKNOWN LOG LEVEL!", ERROR);
```
Comment (RealConrad) [inline src/logger/Logger.cpp]: Remove comment

File: src/main.cpp [hunk 5]
```diff
@@ -3,14 +3,16 @@
 #include <stdexcept>
 
 int main(int argc, char** argv) {
+	Logger::initialize(true);
+	// Logger::hide(Logger::INFO);
+	INFO("Starting program");
     if (argc > 2) {
         std::cerr << "Error: Invalid number of parameters" << std::endl;
```
Comment (RealConrad) [inline src/main.cpp]: Add new Macro style of logs, not `std::cerr`

File: src/main.cpp [hunk 6]
```diff
@@ -3,14 +3,16 @@
 #include <stdexcept>
 
 int main(int argc, char** argv) {
+	Logger::initialize(true);
+	// Logger::hide(Logger::INFO);
```
Comment (RealConrad) [inline src/main.cpp]: Remove comment

**Review by RealConrad — CHANGES_REQUESTED** [review]

File: src/config-manager/ConfigManager.cpp (hunk 1 above)
Comment (kglebows) [reply src/config-manager/ConfigManager.cpp]: I will just rewrite it to new style

**Review by RealConrad — APPROVED** [review]
LGTM!

**Pull request merged**